velox/dwio/parquet/reader/TimestampColumnReader.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/dwio/parquet/reader/IntegerColumnReader.h"
+#include "velox/dwio/parquet/reader/ParquetColumnReader.h"
+
+namespace facebook::velox::parquet {
+
+class TimestampColumnReader : public IntegerColumnReader {
+ public:
+  TimestampColumnReader(
+      const std::shared_ptr<const dwio::common::TypeWithId>& requestedType,
+      std::shared_ptr<const dwio::common::TypeWithId> fileType,
+      ParquetParams& params,
+      common::ScanSpec& scanSpec)
+      : IntegerColumnReader(requestedType, fileType, params, scanSpec) {}
+
+  bool hasBulkPath() const override {
+    return false;
+  }
+
+  void read(
+      vector_size_t offset,
+      RowSet rows,
+      const uint64_t* /*incomingNulls*/) override {
+    auto& data = formatData_->as<ParquetData>();
+    // Use int128_t as a workaroud. Timestamp type in Velox is comprised of an
+    // int64_t seconds_ field and a uint64_t nanos_ field, a total of 16-byte
+    // length
+    prepareRead<int128_t>(offset, rows, nullptr);
```

Comment:
This I can change from `int128_t` to ` Timestamp`. It will require some other changes also. This will help timestamp predicate pushdown to Scan in Gluten. Should I do this in this PR itself? Or raise a followup once this is merge?

Reply:
https://github.com/apache/incubator-gluten/issues/5364#issuecomment-2065712440

Reply:
@mskapilks I think a follow up PR would be better if this implementation doesn't meet the same problem as mention https://github.com/facebookincubator/velox/pull/4680#issuecomment-1527089267

Reply:
@mskapilks I tried the change internally, it seems using int128_t here hits the [NYI](https://github.com/facebookincubator/velox/blob/main/velox/dwio/common/IntDecoder.h#L441) in IntDecoder for plain encoding. It looks to me changing to Timestamp here will fix this case?

Reply:
Is it happening with this PR? There is unit test which read both dictionary and plan in encoding those passes. Custom timestamp decoder is used here so it won't go to that code path. There is other PR for INT96 timestamp there in plain encoded you might face issue.